File: leetcode/python/230.二叉搜索树的第K小的数.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        ret = []
        def middle(node):
            if not node:
                return
            middle(node.left)
            ret.append(node.val)
            middle(node.right)
        middle(root)
        return ret[k - 1]

File: leetcode/golang/230.二叉搜索树的第K小的数.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 var ret []int;
 func kthSmallest(root *TreeNode, k int) int {
	 ret = nil;
	 middle(root);
	 return ret[k - 1];
 }
 
 func middle(root *TreeNode) {
	 if root != nil {
		 middle(root.Left);
		 ret = append(ret, root.Val);
		 middle(root.Right);
	 }
 }